15chrjef/ico-wizard
# [Refactor] Update Step Two

### Summary
This PR changes a lot of components simultaneously to move away from using states. It also deletes a ton of unneeded things, and thorough testing will be required. 

### Things Done
- InputField
  - Deleted dependency of the state's value, and updated the value to be the prop's value. 
  - This whole component needs to be refactored. An inputField should not have its own state, and should only be dependent upon props. 
- ReservedTokenInputBlock
  - Removed all instances of state. 
  - Injected proper stores
  - Update the rendered components to only rely on props from the store. Now, every time a user types something in, it will immediately add it to the store. This is done in the function 'updateReservedTokenInput()'. 
  - Updated method of adding reserved tokens
     - Deleted unneeded code, started using mobx actions/functions instead to simplify. 
  - Created a RenderReservedToken function
    - We do not need to save the React components of reserved tokens. We just need to display them. This function does so by creating and returning an array of components. 
- ReservedTokenItem 
  - Removed instances of state
  - Removed that control statement. It is not needed, and should be done in the parent component.
- StepOne/index.js
  - Passed in the Web3Store
  - Removed state
- StepTwo/index.js
  - Removed State
  - Deleted changeState. This function will be broken down into a series of functions to simplify the code. Methods relating to stateTwo that were previously done here have their own functions that only corresponds to tokens. (updateTokenStore)
  - Deleted ValidateAllFields. This will be done immediately when a user types something. 
  - RenderLinkComponent was simplified.
  - Render does not rely on states anymore, only on props. 
    - ReservedTokenInputBlock does not need to have anything passed into it. 
      - However, if you'd prefer we define the onChange method inside this file instead of in the component, we could do so. 
    - Removed the blur function. Won't be needed now that everything is always being validated.
- ReservedTokenInputStore
  - Added a method to clear the input. This is done when a user adds a reserved token. 
- ReservedTokenStore
  - Added a find method, that will check if a token already exists. 
- TokenStore
  - Computed method written for token validation.
  - Function to validate tokens. 
  - Function to invalidate tokens. This is when a user presses next on stepTwo but hasn't typed anything in yet. 
- Utils
  - Exported functions that are used to validate 
- Other 
  - Trying to fight all the spacing errors. Someone's editor is set to four spaces at all times. That explains all the other files I edited. 
### Testing
- Testing the flow to make sure everything is in order. This PR obviously can't be merged until everything is moved to Mobx, but the stores should contain the proper information .
- Check validation of inputs. Type improper stuff in like a name over 30 characters or a ticker over 3 letters. Should catch it immediately. 
  - Try not putting anything in, and pressing next. Should prevent you. 
- Test adding reserved tokens
  - Try to add two of the same one, you shouldn't be able to. 

### Moving Forward
- We won't need ReservedTokenElementStore. There is no need to store the elements. 
- Merging with the upstream is going to be rough. 
- Update submodules before merging.
